OpenZWave add-on version 0.4.0 (#1394)
home-assistant/hassio-addons: zwave/DOCS.md

# Home Assistant Add-on: OpenZWave

## Installation

Follow these steps to get the add-on installed on your system:

1. Navigate in your Home Assistant frontend to **Supervisor** -> **Add-on Store**.
2. Find the "OpenZWave" add-on and click it.
3. Click on the "INSTALL" button.
4. This add-on currently requires to have the Mosquitto add-on installed.
   Please make sure to install and set up that add-on before continuing.

## How to use

The add-on needs to know where your Z-Wave stick can be found, and therefore,
you'll need to configure the add-on to point to the right device.

If you're using Home Assistant you may find the correct value for this on the
`Supervisor -> System -> Host system -> Hardware` page. It is recommended
to use a "by-id" path to the device if one exists, as it is not subject to
change if other devices are added to the system.

1. Replace `null` in the `device` option in the add-on configuration and specify

>    the device name in quotes: e.g. something like

Reviewer: ```suggestion
   the device name in quotes: e.g., something like
```

instance: 1
```

### Option `device`

The device address of your Z-Wave controller.

If you're using Home Assistant you may find the correct value for this on the
`Supervisor -> System -> Host system -> Hardware` page. It is recommended
to use a "by-id" path to the device if one exists, as it is not subject to
change if other devices are added to the system.

In most cases this looks like one of the following:

- `"/dev/serial/by-id/usb-0658_0200-if00"`
- `"/dev/ttyUSB0"`
- `"/dev/ttyAMA0"`
- `"/dev/ttyACM0"`

### Option `network_key`

Security Z-Wave devices require a network key before being added to the network.
You must set the `network_key` configuration option to use a network key before
adding these devices.

To generate a network key, you can use the following script in, e.g., the SSH
add-on:

```bash
cat /dev/urandom | tr -dc '0-9A-F' | fold -w 32 | head -n 1 | sed -e 's/\(..\)/0x\1, /g' -e 's/, $//'
```

You can also use sites like this one to generate the required data,
just remember to put `0x` before each pair of characters:

<https://www.random.org/cgi-bin/randbyte?nbytes=16&format=h>

Ensure you keep a backup of this key. If you have to rebuild your system and

> don’t have a backup of this key, you won’t be able to reconnect to any security

Reviewer: ```suggestion
don't have a backup of this key, you won't be able to reconnect to any security
```

devices. This may mean you have to do a factory reset on those devices, and your
controller, before rebuilding your Z-Wave network.

### Option `instance` (optional)

Z-Wave instance number as reported to MQTT. This corresponds with the
`instance_id` attribute of `ozw` services in Home Assistant.

The instance ID defaults to `1`, which is generally fine to keep and use.
Only change this in case you are using multiple instances on the same MQTT
server.

## Accessing the ozw-admin application

The add-on allows you to access the underlying ozw-admin application running
in this add-on. This allows you to view, configure and control your Z-Wave
network and its devices.

### Accessing the ozw-admin application via Ingress

You can access the built-in ozw-admin application directly via Home Assistant
Ingress. Click on the "OPEN WEBUI" button to access it.

> If is does not auto connect to your Z-Wave network automatically:

Reviewer: ```suggestion
If it does not auto connect to your Z-Wave network automatically:
```
